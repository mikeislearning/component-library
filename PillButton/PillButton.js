import React from 'react';
import { View, Text, Image, TouchableOpacity } from 'react-native';
import { PropTypes } from 'prop-types';

import styles from './PillButton.styles.js';

export default class PillButton extends React.Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    iconPath: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    isSelected: PropTypes.bool,
    isFixedWidth: PropTypes.bool,
    onPress: PropTypes.oneOfType([PropTypes.string, PropTypes.function]),
  };

  render() {
    const { name, iconPath, isSelected, isFixedWidth, onPress } = this.props;
    const pillButtonStyle = [styles.pillButtonContainer];
    const textStyle = [styles.text];
    if (isSelected) {
      pillButtonStyle.push(styles.selectedPillButton);
      textStyle.push(styles.selectedText);
    }

    const pillButtonContent = (
      <View style={pillButtonStyle} width={isFixedWidth ? 130 : 'auto'}>
        {iconPath && <Image style={styles.image} source={iconPath} />}
        <Text style={textStyle}>{name}</Text>
      </View>
    );

    return (
      <React.Fragment>
        {onPress ? (
          <TouchableOpacity onPress={onPress}>{pillButtonContent}</TouchableOpacity>
        ) : (
          pillButtonContent
        )}
      </React.Fragment>
    );
  }
}
