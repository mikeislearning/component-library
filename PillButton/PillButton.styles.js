import { StyleSheet } from 'react-native';
import { colors } from 'app/theme';

export default StyleSheet.create({
  pillButtonContainer: {
    backgroundColor: colors.white,
    borderRadius: 30,
    borderColor: colors.gray.medium,
    borderWidth: 1,
    margin: 5,
    padding: 16,
    alignSelf: 'center',
  },
  selectedPillButton: {
    backgroundColor: colors.green.light,
    borderColor: colors.green.dark,
    borderWidth: 2,
    padding: 15,
  },
  image: {
    width: 20,
    height: 20,
    marginRight: 10,
  },
  text: {
    color: colors.gray.medium,
    fontSize: 14,
    alignSelf: 'center',
  },
  selectedText: {
    color: colors.black,
    fontWeight: '600',
  },
});
