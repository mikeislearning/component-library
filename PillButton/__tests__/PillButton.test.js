/**
 * @format
 */

import React from 'react';
import { render } from 'react-native-testing-library';

import PillButton from '../PillButton';

describe('<PillButton />', () => {
  it('renders correctly', () => {
    const pillButton = render(<PillButton name={'name'} iconPath={5} />);
    expect(pillButton).toMatchSnapshot();
  });
});
