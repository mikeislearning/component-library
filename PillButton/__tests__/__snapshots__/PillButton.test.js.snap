// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PillButton /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#726E67",
        "borderRadius": 30,
        "borderWidth": 1,
        "margin": 5,
        "padding": 16,
      },
    ]
  }
  width="auto"
>
  <Image
    source={5}
    style={
      Object {
        "height": 20,
        "marginRight": 10,
        "width": 20,
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "color": "#726E67",
          "fontSize": 14,
        },
      ]
    }
  >
    name
  </Text>
</View>
`;
